org.bardsoftware.eclipsito/src/org/bardsoftware/eclipsito/Boot.java

```diff
@@ -173,8 +173,8 @@ public static void main(String args[]) {
     }
     String[] pathParts = modules.split(File.pathSeparator);
     for (int i = 0; i < pathParts.length; i++) {
-      if(pathParts[i].startsWith("~")) {
-        pathParts[i] = pathParts[i].replaceAll("~", System.getProperty("user.home"));
+      if (pathParts[i].startsWith("~")) {
+        pathParts[i] = pathParts[i].replace("~", System.getProperty("user.home"));
```

Comment:
replace and replaceAll both [replace all matches](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replace(java.lang.CharSequence,%20java.lang.CharSequence)). The difference between them is that the latter accepts regexp.
Old good substring and concatenation should do the trick. Please write the tests :)

Reply:
Ok, but what about replaceFirst?